coral-hive/src/test/java/com/linkedin/coral/common/IcebergTableConverterTest.java

```diff
@@ -0,0 +1,173 @@
+
+    testCatalog.addTable(TEST_DB, TEST_TABLE, icebergCoralTable);
+
+    // Create converter with test catalog
+    converter = new HiveToRelConverter(testCatalog);
+  }
+
+  /**
+   * Creates a mock Iceberg table with a timestamp(6) column using Mockito.
+   * Schema: (id BIGINT, event_time TIMESTAMP, name STRING)
+   */
+  private org.apache.iceberg.Table createMockIcebergTable() {
+    // Define Iceberg schema with timestamp column
+    // TimestampType.withoutZone() represents microsecond precision (6 digits)
+    Schema icebergSchema = new Schema(Types.NestedField.required(1, "id", Types.LongType.get()),
+        Types.NestedField.required(2, "event_time", Types.TimestampType.withoutZone()),
+        Types.NestedField.optional(3, "name", Types.StringType.get()));
+
+    // Mock the Iceberg Table interface using Mockito
+    org.apache.iceberg.Table mockTable = mock(org.apache.iceberg.Table.class);
+
+    // Stub necessary methods
+    when(mockTable.schema()).thenReturn(icebergSchema);
+    when(mockTable.properties()).thenReturn(Collections.emptyMap());
+    when(mockTable.name()).thenReturn(TEST_DB + "." + TEST_TABLE);
+
+    return mockTable;
+  }
+
+  /**
+   * Test that an Iceberg table with TIMESTAMP column is correctly converted to RelNode
+   * with precision 6 (microseconds).
+   */
+  @Test
+  public void testIcebergTableWithTimestampPrecision() {
+    // Convert SQL query to RelNode
+    String sql = String.format("SELECT * FROM %s.%s", TEST_DB, TEST_TABLE);
+    RelNode relNode = converter.convertSql(sql);
+
+    // Print RelNode structure for debugging
+    System.out.println("\nField types in RelNode:");
+    for (RelDataTypeField f : relNode.getRowType().getFieldList()) {
+      System.out.println(f.getName() + ": " + f.getType());
+    }
+
+    System.out.println("\nRelNode structure:");
+    System.out.println(org.apache.calcite.plan.RelOptUtil.toString(relNode));
+
+    // Verify RelNode structure
+    assertNotNull(relNode, "RelNode should not be null");
+
+    // Check that we have 3 columns: id, event_time, name
+    List<RelDataTypeField> fields = relNode.getRowType().getFieldList();
+    assertEquals(fields.size(), 3, "Should have 3 columns");
+
+    // Verify column names
+    assertEquals(fields.get(0).getName(), "id");
+    assertEquals(fields.get(1).getName(), "event_time");
+    assertEquals(fields.get(2).getName(), "name");
+
+    // Verify column types
+    assertEquals(fields.get(0).getType().getSqlTypeName(), SqlTypeName.BIGINT);
+    assertEquals(fields.get(2).getType().getSqlTypeName(), SqlTypeName.VARCHAR);
+
+    // CRITICAL: Verify that TIMESTAMP has precision 6 (microseconds)
+    RelDataType timestampType = fields.get(1).getType();
+    assertEquals(timestampType.getSqlTypeName(), SqlTypeName.TIMESTAMP, "event_time should be TIMESTAMP");
+    assertEquals(timestampType.getPrecision(), 6, "Timestamp should have precision 6 (microseconds)");
+  }
```

Comment:
I found [here](https://github.com/apache/calcite/blob/63633ac9fba4bfc9538f850a08a283c1f7279fec/core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystemImpl.java#L204) that the max precision of `TIMESTAMP` in Calcite is `3`. I also wrote a similar test in my local and it failed for me. Curious how it worked here. Could you clarify which particular `RelDataTypeFactory` is being picked up in this test?